.github/workflows/build-and-deploy.yaml

```diff
@@ -0,0 +1,126 @@
+    tags: ["v*.*.*"]
+    paths-ignore:
+      - doc/**
+      - samples/**
+      - README.md
+      - CONTRIBUTING.md
+
+env:
+  # Use docker.io for Docker Hub if empty
+  REGISTRY: ghcr.io
+  # github.repository as <account>/<repo>
+  IMAGE_NAME: ${{ github.repository }}
+  RBAC_FILE_PATH: config/rbac/role.yaml
+  CRD_FILE_PATH: config/crd/skiperator.kartverket.no_applications.yaml
+
+jobs:
+  build:
+    name: Build container image
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log into registry ${{ env.REGISTRY }}
+        uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          tags: type=sha,format=long
+
+      - name: Build and push Docker image
+        id: build-docker
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          push: ${{ !github.event.pull_request.draft }}
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
+  generate:
+    name: CRD and ClusterRole
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Setup Golang environment
+        uses: actions/setup-go@v4
+        with:
+          go-version: "1.20"
+
+      - name: Generate CRD and ClusterRole
+        run: make generate
+
+      - name: Upload CRD and ClusterRole
+        uses: actions/upload-artifact@v3
+        with:
+          name: config
+          path: |
+            ${{ env.RBAC_FILE_PATH }}
+            ${{ env.CRD_FILE_PATH }}
+
+  deploy-argo:
+    if: github.event_name == 'push' || github.ref == 'refs/heads/main'
+    needs: [build, generate]
+    runs-on: ubuntu-latest
+    env:
+      BASE_DIR: ./bases/skiperator
+    steps:
+      - name: Checkout apps repo
+        uses: actions/checkout@v3
+        with:
+          repository: kartverket/skip-apps
+          token: ${{ secrets.SKIPERATOR_DEPLOY_SECRET }}
+
+      - name: Download CRD and RBAC
+        uses: actions/download-artifact@v3
+        with:
+          name: config
+
+      - name: Patch Image Digest
+        run: |
+          kubectl patch --local \
+            -f $BASE_DIR/kustomize.yaml \
+            -p '{"images":[{"name":"${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}","digest":"${{needs.build.outputs.image_digest}}"}]}' \
+            -o yaml
+
+      - name: Update CRD and Role
+        run: |
+          cp -f $CRD_FILE_PATH $BASE_DIR/crd.yaml
+          cp -f $RBAC_FILE_PATH $BASE_DIR/clusterrole.yaml
```

Comment:
En nice to have på alle cp-er er å legge på -v så operasjonen skrives til byggloggen